app/models/mixins/archived_mixin.rb

```diff
@@ -2,11 +2,14 @@ module ArchivedMixin
   extend ActiveSupport::Concern
 
   included do
+    scope :deleted, -> { where(:deleted => true) }
+    scope :not_deleted, -> { where(:deleted => false) }
```

Comment:
Consider first adding these scopes (with current ems_id/deleted_on implementation), switching to them everywhere (including adding the implicit `not_deleted` where needed), and only then migrating to the new implementation.